### ISSUE TITLE
[FEATURE] Adding persian nemo ASR models

### ISSUE THREAD

**Issue by mah92** [k2-fsa/sherpa-onnx#3714]
### Describe the feature
I have tested these brand new nemo models for persian (farsi) ASR and found them specially accurate,

https://huggingface.co/Reza2kn/Shenava-Koochik-v1.0

https://huggingface.co/Reza2kn/Shenava-Rizeh-v1.0

https://huggingface.co/Reza2kn/Shenava-Rizeh-Pizeh-v1.0

### Motivation / Use Case
After semi-accurate vosk models, there is no zipformer or any kind of ASR models for persian implemented in sherpa onnx. Whisper models are not accurate specially in small forms, usable for edge devices.

| Member | golden‑6669 WER | CER | FLEURS‑fa WER | CER |
|---|---|---|---|---|
| Koochik v1.0 (114M) | **7.49%** | 2.30% | **10.64%** | 3.79% |
| Rizeh v1.0 (32M) | 12.11% | 3.94% | 14.45% | 5.10% |
| Rizeh Pizeh v1.0 (6.9M) | 24.55% | 8.89% | 26.95% | 10.22% |

### Additional context
The models (Koochik-v1.0) are used and can be tested in shenava.app website.

**Comment by mah92** [comment]
Some time ago, you told me to just create an issue to merge pre-trained models in sherpa. Should I create onnx models based on sherpa docs before you merge?
@csukuangfj

**Comment by csukuangfj** [comment]
Please describe why the listed models cannot be exported and used in sherpa-onnx.

**Comment by mah92** [comment]
Besides my first demand, which was to add these models into sherpa onnx pipeline, I had problem convering them:

Rizeh Pizeh is a pure ctc model, so I could follow [this](https://github.com/k2-fsa/sherpa/blob/master/docs/source/onnx/pretrained_models/offline-ctc/nemo/how-to-export.rst).

But Rizeh and Koochik are RNN/CTC models. They come from [nvidia/stt_fa_fastconformer_hybrid_large](https://huggingface.co/nvidia/stt_fa_fastconformer_hybrid_large) which is advertised as a "large" version of FastConformer Transducer-CTC.

I could not find a direct recipe.

I found these similar sherpa-onnx compatible exports, which have different approaches.

https://huggingface.co/dev-ahmedhany/stt_ar_fastconformer_hybrid_large_streaming_pcd_v1.0-sherpa -> this converts the model into 3 models: decoder.onnx, encoder.onnx, and joiner.onnx

https://huggingface.co/nur-dev/nemo-fast -> this converts the model into 2 models: encoder.onnx and decoder_joint.onnx

Which is the proper method used in sherpa for these? 
online/offline? (These are online models)
ctc/transducer? 
How many onnx files should I generate? 1/2/3?
Is there a direct recepie?

@csukuangfj

**Comment by csukuangfj** [comment]
please see the scripts/nemo

if you have further issues, please post errors when exporting.

**Comment by nshmyrev** [comment]
@mah92 btw, that Koochik model also has only CTC head, RNNT decoding doesn't work.

**Comment by Reza2kn** [comment]
@nshmyrev no, the model actually has both fucntional RNNT and CTC heads, but the uploaded ONNX checkpoint is a CTC-only export, made for browser deployment, not sherpa-onnx. while not necessary here, if you **do** need RNNT for some reason, plz check out the original nemo checkpoints.

@csukuangfj on the reason why the onnx checkpoints couldn't be used as is:

**Why it didn’t work before:** sherpa-onnx’s from_nemo_ctc always applies NeMo per-feature normalization (NemoNormalizeFeatures only implements normalize_type=per_feature). These FastConformer models were trained with normalize=NA (no feature normalization), so the forced normalization turned the output into garbage — and there was no obvious way to turn it off.

**The solution** (no sherpa-onnx code change): export the CTC head to ONNX taking un-normalized log-mel (audio_signal [N,80,T], dynamic T, fp32) and set the ONNX metadata normalize_type to empty (not per_feature). sherpa-onnx then skips normalization and decodes correctly via stock from_nemo_ctc.

**Worth noting for maintainers**: (1) NemoNormalizeFeatures errors on anything but per_feature — empty-string works but is undocumented; **explicitly supporting normalize_type=none would make normalization-free NeMo models first-class.** (2) The **Python** from_nemo_ctc **doesn’t expose is_librosa/window_type/preemph_coeff** that the **C++** FeatureExtractorConfig **has**; these models tolerate the default HTK mel, but **exposing librosa-mel would let them match NeMo’s slaney mel exactly (helps the smallest model)**. (3) **Persian numbers are spelled by design → pair with an ITN post-processor.**

I managed to export proper sherpa-onnx checkpoints for all 3 models:
[Reza2kn/Shenava-Koochik-v1.0-sherpa-onnx](https://huggingface.co/Reza2kn/Shenava-Koochik-v1.0-sherpa-onnx)
[Reza2kn/Shenava-Rizeh-v1.0-sherpa-onnx](https://huggingface.co/Reza2kn/Shenava-Rizeh-v1.0-sherpa-onnx)
[Reza2kn/Shenava-Rizeh-Pizeh-v1.0-sherpa-onnx](https://huggingface.co/Reza2kn/Shenava-Rizeh-Pizeh-v1.0-sherpa-onnx)
and the next batch of models is cooking as well! 🔥🫡

**Comment by nshmyrev** [comment]
@Reza2kn thank you for sharing the great models. About RNNT decoding, could you clarify please, only ctc decoder works here:

```
import nemo.collections.asr as nemo_asr

asr_model = nemo_asr.models.EncDecHybridRNNTCTCBPEModel.restore_from('models/Shenava-Koochik-v1.0/shenava-koochik-v1.0.nemo', strict=False)
# asr_model.change_decoding_strategy(decoder_type='ctc') <- if commented out outputs garbage
transcript = asr_model.transcribe(files)
```

with CTC

```
fl_1660_16142317535120109527 رومانتیسم اثر عظیمی از جب گرایی فرهنگی داشت که از نویسندگان نظیر گوته، فیشته و ش
```

with rnnt (ctc commented out)

```
fl_1660_16142317535120109527 شتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشتهشته بودند صدایططط صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای صدای زمان 
```

**Comment by Reza2kn** [comment]
@nshmyrev  oh my bad! turns out i had the wrong impressions for this v1 model. it seems like the RNNT IS really broken here, in fact.
out of curiosity, may i ask what;s your use case with it? error correction or something else?

**Comment by Reza2kn** [comment]
@nshmyrev while still _not_ the final model, here's an int8 quant of an RNNT sherpa-onnx model that **does** work:

[Reza2kn/Shenava-Koochik-v1.5-RNNT-sherpa-onnx](https://huggingface.co/Reza2kn/Shenava-Koochik-v1.5-RNNT-sherpa-onnx)

**Comment by mah92** [comment]
By the way, I converted, tested and uploaded converted models here:

https://huggingface.co/mah92/sherpa-onnx-shenava-koochik-v1.0-ctc-non-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-koochik-v1.0-ctc-non-streaming-int8

https://huggingface.co/mah92/sherpa-onnx-shenava-koochik-v1.0-ctc-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-koochik-v1.0-ctc-streaming-int8

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-v1.0-ctc-non-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-v1.0-ctc-non-streaming-int8

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-v1.0-ctc-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-v1.0-ctc-streaming-int8

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-pizeh-v1.0-ctc-non-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-pizeh-v1.0-ctc-non-streaming-int8

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-pizeh-v1.0-ctc-streaming

https://huggingface.co/mah92/sherpa-onnx-shenava-rizeh-pizeh-v1.0-ctc-streaming-int8

If you would, please add these models into sherpa-onnx framework, so that they become easier to access. I think this issue is over. 
@csukuangfj

**Comment by csukuangfj** [comment]
Please contribute by updating
https://github.com/k2-fsa/sherpa-onnx/blob/master/.github/workflows/upload-models.yaml
if you want to add them to sherpa-onnx

By the way, please embed the language in the folder name. Otherwise, it is not clear what language the model supports by just looking at the name.

**Comment by mah92** [comment]
I did my best. @csukuangfj 
    PR: #3825
